### PR TITLE
feat: donate button show amount

### PR DESCRIPTION
**Pull request by michael-odonovan**
#### What is it doing?
Adds the amount to be donated to the button when the moneybuys or custom amount are inputed / selected.

#### Why is this required?
FR / BBC around RND23

#### link to Jira ticket:
https://comicrelief.atlassian.net/browse/ENG-2353

### Quick Checklist:
- [x] My PR title follows the Conventional Commit spec.

- [x] I have filled out the PR description as per the template above.

- [x] I have added tests to cover new or changed behaviour.

- [x] I have updated any relevant documentation.

### CODE REVIEW

**Review by AndyEPhipps — CHANGES_REQUESTED** [review]
Looks good to me, man- just two things:

While we're here, do you think we could update the button font to `inherit` the proper font-family?

Also, can you see what Curtis reckons of this? I feel like there's a nicer message to show here rather than just omitting the value:

Screenshot 2023-04-06 at 14 52 14

**Comment by michael-odonovan** [comment]
ok cool, will do

**Comment by michael-odonovan** [comment]
- Curtis suggested that when the input field is left blank / cleared, the click outside should return the money buy to the default.
- I also removed the lonely "£" when no value is entered.

File: src/components/Organisms/Donate/Form/useClickOutside.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        setAmountDonate(parseFloat(thisAmount));
+      }
+    }
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside);
+    };
+  }, [amountRef, errorMsg]);
```
Comment (AndyEPhipps) [inline src/components/Organisms/Donate/Form/useClickOutside.js:40]: Can you add these useEffect deps in?

File: src/components/Organisms/Donate/Form/useClickOutside.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  givingType,
+  singleGiving,
+  errorMsg,
+  regularGiving
+) {
+  useEffect(() => {
+    /**
+     * Alert if clicked on outside of element
+     */
+    function handleClickOutside(event) {
```
Comment (AndyEPhipps) [inline src/components/Organisms/Donate/Form/useClickOutside.js:16]: Hmm could we move the function declaration out of the useEffect so we're not recreating  it unnecessarily each time?

File: src/components/Organisms/Donate/Form/useClickOutside.js (hunk 1 above)
Comment (AndyEPhipps) [reply src/components/Organisms/Donate/Form/useClickOutside.js:40]: Hmmm if you're binding the function to the mouseup and down functions, why does it need to be in a useEffect, retriggering when amountRef or errorMsg gets updated?

File: src/components/Organisms/Donate/Form/useClickOutside.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+          : givingData.moneybuys[0].value;
+
+        setAmountDonate(parseFloat(thisAmount));
+      }
+    }
+    // Bind the event listener
+    document.addEventListener('mousedown', handleClickOutside);
+    return () => {
+      // Unbind the event listener on clean up
+      document.removeEventListener('mousedown', handleClickOutside);
```
Comment (AndyEPhipps) [inline src/components/Organisms/Donate/Form/useClickOutside.js:38]: V nice

**Comment by michael-odonovan** [comment]
@AndyEPhipps 
Fiddled around with this a lot yesterday. 
- have moved the code back into the main Form.js file.
- the latest commit has the handleClickOutside function outside of the useEffect, however it will complain / error unless you wrap it in the useCallback hook.
- the previous commit has the handleClickOutside function inside the useEffect, and this obvs also works.
I'll leave it to you to decide which way to go.
I also added better button colours to the markdown examples as I couldn't see the red writing with the red background (this is on master as well).

**Comment by AndyEPhipps** [comment]
Assigning Paul to review this too, so I'm not blocking it while I'm tied-up

**Comment by michael-odonovan** [comment]
ok cool

**Comment by pauldolden** [comment]
Everything seems to work here as expected here. Only question is about the snapshots, they appear to render the null values. Nothing that appears to affect the behaviour, but worth mentioning.

**Comment by michael-odonovan** [comment]
> Only question is about the snapshots, they appear to render the null values. 

Cheers Paul. However I don't understand this bit. I can see src/components/Organisms/Donate/__snapshots__/Donate.test.js.snap, the jsx being changed by Jest/snapshots. But I don't understand null values?

**Comment by pauldolden** [comment]
> > Only question is about the snapshots, they appear to render the null values.
> 
> Cheers Paul. However I don't understand this bit. I can see src/components/Organisms/Donate/**snapshots**/Donate.test.js.snap, the jsx being changed by Jest/snapshots. But I don't understand null values?

Hey Mike, apologies for the confusion here:
```
Donate £  now
```

I just meant that these are being rendered as per the above in the snapshots, just a comment more than anything else.

**Comment by michael-odonovan** [comment]
oh yeah gotcha now => I suppose Jest can't parse the Js conditionals and just grabs from a state somewhere along the line?
I recon prob not worth messing with it as we just use snaps to check things haven't changed suddenly.
What you recon?

**Comment by michael-odonovan** [comment]
Have changed null to 10 in the initial useState for the donation value, which has sorted this issue.

**Comment by pauldolden** [comment]
> Have changed null to 10 in the initial useState for the donation value, which has sorted this issue.

Cool, I'm happy to approve this. All seems to work as expected. Unless @AndyEPhipps and @KrupaPammi have anything else to add?

**Comment by michael-odonovan** [comment]
Have sorted out Button issue. Error message is in view.

**Comment by AndyEPhipps** [comment]
> Tried to use setFocus() on the input after clicking the button, but something isn't working, so looks quite tricky to get that

I'll give this a look as part of the re-review, hold on your hat

**Review by AndyEPhipps — APPROVED** [review]
Good stuff, mate 👍

**Review by AndyEPhipps — CHANGES_REQUESTED** [review]
Although it looks like you've remove recent work with that master merge; can re-do this please

**Pull request merged**